src/ert/config/external_ert_script.py

```diff
@@ -20,22 +24,39 @@ def __init__(self, ert: EnKFMain, storage: StorageAccessor, executable: str):
         self.__job: Optional[Popen[bytes]] = None
 
     def run(self, *args: Any) -> None:
-        command = [self.__executable]
-        command.extend([str(arg) for arg in args])
+        script_content = (
+            Path(self.__executable).read_text(encoding="utf-8").splitlines()
+        )
+        tmp_script_path: Optional[str] = None
+        if self.__executable[-2:] == "sh":
+            script_content.insert(1, "set -e")
+        try:
+            with tempfile.NamedTemporaryFile(
+                mode="w+", encoding="utf-8", delete=False
+            ) as tmp_file:
+                tmp_file.write("\n".join(script_content))
+                tmp_file.flush()
+                tmp_script_path = tmp_file.name
+            os.chmod(tmp_script_path, os.stat(tmp_script_path).st_mode | stat.S_IEXEC)
+            command = [tmp_script_path]
+            command.extend([str(arg) for arg in args])
 
-        # we take care to terminate the process in cancel()
-        self.__job = Popen(command, stdout=PIPE, stderr=PIPE)
+            # we take care to terminate the process in cancel()
+            self.__job = Popen(command, stdout=PIPE, stderr=PIPE)
 
-        # The job will complete before stdout and stderr is returned
-        stdoutdata, stderrdata = self.__job.communicate()
+            # The job will complete before stdout and stderr is returned
+            stdoutdata, stderrdata = self.__job.communicate()
 
-        self._stdoutdata = codecs.decode(stdoutdata, "utf8", "replace")
-        self._stderrdata = codecs.decode(stderrdata, "utf8", "replace")
+            self._stdoutdata = codecs.decode(stdoutdata, "utf8", "replace")
+            self._stderrdata = codecs.decode(stderrdata, "utf8", "replace")
 
-        sys.stdout.write(self._stdoutdata)
+            sys.stdout.write(self._stdoutdata)
 
-        if self.__job.returncode != 0:
-            raise RuntimeError(self._stderrdata)
+            if self.__job.returncode != 0:
+                raise RuntimeError(self._stderrdata)
+        finally:
```

Comment:
This cleanup is a bonus fix, or is it related to the fix of the bug?

Reply:
It is part of the fix. I create a copy of the script file, add the `set -e`-option , run it, then delete it at the end.

Reply:
Got it. 

What is  self.__executable is a binary file (I do not know if that is disallowed already), could we break the functionality for that with this?

Reply:
I check the file extension (or rather if the filename ends with sh - this could maybe cause problems), but maybe there is a better way. @xjules and I tried to set the flag in subprocess.Popen, but that was not possible.